collect_app/src/main/java/org/odk/collect/android/activities/GoogleSheetsUploaderActivity.java

```diff
@@ -122,22 +120,16 @@ protected void onCreate(Bundle savedInstanceState) {
     }
 
     private void runTask() {
-        instanceGoogleSheetsUploaderTask = (InstanceGoogleSheetsUploaderTask) getLastCustomNonConfigurationInstance();
```

Comment:
Using `getLastCustomNonConfigurationInstance()` comes from very first versions of this functionality when we allowed calling this part of code again. Now it's called only from `onCreate()` but in `onDestroy()` we have `finish()` that means `onCreate()` can't be called again according to the lifecycle:
https://developer.android.com/guide/components/activities/activity-lifecycle